src/functions/indGraphFat.jl

```diff
@@ -0,0 +1,58 @@
+# This implements the variant when A is dense and m < n
+
+struct IndGraphFat{T <: RealOrComplex} <: IndGraph
+  m::Int
+  n::Int
+  A::AbstractArray{T,2}
+  AA::AbstractArray{T,2}
+  F::Base.LinAlg.Cholesky #LDL factorization
+end
+
+function IndGraphFat(A::AbstractArray{T,2}) where {T <: RealOrComplex}
+  m, n = size(A)
+  AA = A * A'
+
+  F = LinAlg.cholfact(eye(m) + AA)
+  #normrows = vec(sqrt.(sum(abs2.(A), 2)))
+
+  IndGraphFat(m, n, A, AA, F)
+end
+
+# is_convex(f::IndGraph) = true
+# is_set(f::IndGraph) = true
+# is_cone(f::IndGraph) = false
+
+function prox!{T <: RealOrComplex}(
+    x::AbstractArray{T, 1},
+    y::AbstractArray{T, 1},
+    f::IndGraphFat,
+    c::AbstractArray{T, 1},
+    d::AbstractArray{T, 1})
+  y[:] = f.F \ (f.A * c + f.AA * d)
```

Comment:
Here it is probably good to have a `tmp` vector allocated in IndGraphFat if we want to avoid allocating memory in prox! , then we could do something like
```julia
A_mul_B!(tmp, f.A, c) # Maybe tmp .= f.A * c works equally well? Try it! 
A_mul_B!(y, f.AA, d)   # Maybe tmp .= f.AA * d
tmp += y
A_ldiv_B(y, f.F, tmp)
```